test/node-ranges.js

```diff
@@ -0,0 +1,85 @@
+const { parse, HTMLElement, TextNode, CommentNode } = require('../dist');
```

Comment:
This test parses with `htmlparser2` and compares ranges and nodes. Although it serves as a test for range comparison, it will also be helpful in catching a broad range of other issues with the parser that present tests may not cover. This is because it ensures the AST structures are the same for all fundamental node types (blockText, comment, self-closing, etc) parsed by the parser.